app/src/main/java/net/gsantner/markor/format/markdown/MarkdownActionButtons.java

```diff
@@ -41,7 +41,7 @@ public class MarkdownActionButtons extends ActionButtonBase {
 
     private static final Pattern WEB_URL = Pattern.compile("https?://[^\\s/$.?#].[^\\s]*");
 
-    private final Set<Integer> _disabledHeadings = new HashSet<>();
+    private final MarkorDialogFactory.HeadlineDialogState _headlineDialogState = new MarkorDialogFactory.HeadlineDialogState();
```

Comment:
@guanglinn Instead of inserting arbitrary data into the intent, we create and pass around a state class for the headline dialog